tests/Console/Report/ListSetsReport/JsonReporterTest.php

```diff
@@ -38,7 +38,7 @@ protected function getFormat()
     protected function assertFormat($expected, $input): void
     {
         static::assertJsonSchema($input);
-        static::assertJsonStringEqualsJsonString($expected, $input);
+        static::assertSame($expected, $input);
```

Comment:
why the change? comparing `assertJsonStringEqualsJsonString` over `assertSame` is nice, as `assertJsonStringEqualsJsonString` will ignore JSON-irrelevant whitespaces

Reply:
`assertSame` is more strict, as `assertJsonStringEqualsJsonString` normalizes hashes so it doesn't test if elements are in order that is expected

Reply:
order of element in json hash should not be relevant. why you think it is relevant for us?
(remember it's JSON, not PHP assoc array that is keeping the order of keys)

Reply:
I assumed it matter because of the sorting here:
https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/v3.0.1/src/Console/Report/ListSetsReport/JsonReporter.php#L42
which I was changes and wanted to test

Reply:
the assoc-array we have may be having sorted items, but after we convert to json with `json_encode`, we have no control whether the JSON is having sorted keys or not.
some implementations are explicitly aiming that, eg https://www.npmjs.com/package/json-stable-stringify for node, but I'm not aware about any promise like that in PHP.

then, whoever reads JSON generated by us, they may not keep the order we produced anyway, eg if read by perl, the order of keys in hash is undefined.

Reply:
(also, `assertJsonStringEqualsJsonString` should have better error message than `assertSame` when showing diff for 2 JSONs)

Reply:
ok, should I remove the sorting from report builders as these hold no value? would be nice clean up :)

Reply:
sorting in builder was helping me debugging whether it generates content properly, I would keep it tbh ;)

Reply:
that gets us in the odd spot that I cannot fix it because you don't like the fix to be tested ;)

Reply:
restored, the output generators don't have to sort it seems as the source already sorts:
https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/v3.0.1/src/RuleSet/RuleSets.php#L45
but I left as is for now